### PR TITLE
Converter slang

### PR DESCRIPTION
**Pull request by marcelo-sousa-sonarsource**
Initial converter to SLang binary expressions.

### CODE REVIEW

File: slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangParserTest.java [hunk 1]
```diff
@@ -42,4 +51,20 @@ public void testFile() throws IOException {
...
+    SLangParser parser = new SLangParser(tokens);
+    SLangParser.SlangFileContext tree = parser.slangFile();
+    ParseTreeWalker walker = new ParseTreeWalker();
+
+    for (Token token : tokens.getTokens()) {
+      System.out.println(token.getLine() + ": " + token.getText());
+    }
+
+    assertThat(tree.children, notNullValue());
+    assertThat(tree.children.isEmpty(), is(false));
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangParserTest.java]: we switched to assertj for assertion to be more verbose

File: slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+              .collect(toList());
+      return new NativeTreeImpl(null, new SNativeKind(), children);
+    }
+
+    @Override
+    public Tree visitDisjunction(SLangParser.DisjunctionContext ctx) {
+      int conjunctionSize = ctx.conjunction().size();
+
+      if (conjunctionSize > 1) {
+        if (conjunctionSize > 2) {
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java]: When > 1, it should always return `BinaryExpressionTree` object, with nested nested children `BinaryExpressionTree` depending on the number of children

File: slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Tree visitConjunction(SLangParser.ConjunctionContext ctx) {
+      int equalitySize = ctx.equalityComparison().size();
+
+      if (equalitySize > 1) {
+        if (equalitySize > 2) {
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java]: Same as `visitDisjunction`

File: slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Tree visitEqualityComparison(SLangParser.EqualityComparisonContext ctx) {
+      int comparisonSize = ctx.comparison().size();
+
+      if (comparisonSize > 1) {
+        if (comparisonSize > 2) {
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java]: Same as `visitDisjunction`

File: slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Tree visitComparison(SLangParser.ComparisonContext ctx) {
+      int additiveSize = ctx.additiveExpression().size();
+
+      if (additiveSize > 1) {
+        if (additiveSize > 2) {
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java]: Same as `visitDisjunction`

File: slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Tree visitAdditiveExpression(SLangParser.AdditiveExpressionContext ctx) {
+      int multiplicativeSize = ctx.multiplicativeExpression().size();
+
+      if (multiplicativeSize > 1) {
+        if (multiplicativeSize > 2) {
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java]: Same as `visitDisjunction`

File: slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+
+      return visitChildren(ctx);
+    }
+
+    @Override
+    public Tree visitMultiplicativeExpression(SLangParser.MultiplicativeExpressionContext ctx) {
+      int atomicExpressionSize = ctx.atomicExpression().size();
+
+      if (atomicExpressionSize > 1) {
+        if (atomicExpressionSize > 2) {
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/main/java/com/sonarsource/slang/parser/SLangConverter.java]: Same as `visitDisjunction`

File: slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangConverterTest.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+
+    TreeVisitor<TreeContext> visitor = new TreeVisitor<>();
+
+    visitor.register(BinaryExpressionTreeImpl.class, (ctx, binaryExpressionTree) -> numBinNodes.getAndIncrement());
+    visitor.register(IdentifierImpl.class, (ctx, identifierTree) -> numIdentifierNode.getAndIncrement());
+    visitor.register(LiteralTreeImpl.class, (ctx, literalTree) -> numLiteralNode.getAndIncrement());
+    visitor.scan(new TreeContext(), tree);
+
+    assertThat(numBinNodes.get(), is(6));
+    assertThat(numIdentifierNode.get(), is(7));
```
Comment (christophe-zurn-sonarsource) [inline slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangConverterTest.java]: we switched to assertj for assertion to be more verbose

File: slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangParserTest.java (hunk 1 above)
Comment (pynicolas) [reply slang-antlr/src/test/java/com/sonarsource/slang/antlr/SLangParserTest.java]: more "explicit".

**Pull request merged**